### PR TITLE
Feat/storybook language currency

### PR DESCRIPTION
**Pull request by iurimatias**
### What does the PR do

- adds the language & currency settings screen to storybook, refactors it a bit
- note: the currency/language component is something that should also go into storybook but this is not done in this PR

Screen Shot 2022-10-13 at 5 53 12 PM

### CODE REVIEW

File: storybook/main.qml [hunk 1]
```diff
@@ -53,6 +52,24 @@ ApplicationWindow {
                     selected: viewLoader.source.toString().includes(title)
                     onClicked: mainPageView.page(title);
                 }
+
+                StatusNavigationListItem {
+                    title: "AboutView"
```
Comment (caybro) [inline storybook/main.qml:57]: This belongs to another PR

File: storybook/pages/LanguageCurrencySettingsPage.qml [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+
+            currencyStore: QtObject {
+                property string currentCurrency: "usd"
+                property string currentCurrencySymbol: "usd"
+
+                readonly property ListModel currenciesModel: ListModel {
+                    ListElement {
+                        key: "usd"
+                        shortName: "USD"
+                        name: qsTr("US Dollars")
```
Comment (caybro) [inline storybook/pages/LanguageCurrencySettingsPage.qml:65]: No need for qsTr() here

File: ui/StatusQ/src/StatusQ/Controls/StatusBinaryRadioControl.qml [hunk 3]
```diff
@@ -0,0 +1,39 @@
+import QtQuick 2.14
```
Comment (caybro) [inline ui/StatusQ/src/StatusQ/Controls/StatusBinaryRadioControl.qml:1]: We don't need this control at all; standard radio button handles that directly.

Radio buttons are mutually exclusive (binary) by default and if there's a need/pattern for such control, they can be always placed inside a ButtonGroup or GroupBox

File: ui/app/AppLayouts/Profile/ProfileLayout.qml [hunk 4]
```diff
@@ -207,11 +207,26 @@ StatusSectionLayout {
...
+                    }
+
+                    function getReleaseNotes() {
+                        console.log("TODO: getReleaseNotes")
+                    }
+
+                    function openLink(url) {
+                        Global.openLink(url)
+                 }
+                }
```
Comment (caybro) [inline ui/app/AppLayouts/Profile/ProfileLayout.qml:229]: Some weird indentation around here

File: ui/app/AppLayouts/Profile/views/AboutView.qml [hunk 5]
```diff
@@ -7,7 +7,6 @@ import StatusQ.Core.Theme 0.1
 import StatusQ.Controls 0.1
```
Comment (caybro) [inline ui/app/AppLayouts/Profile/views/AboutView.qml:7]: Belongs to another PR

**Review by caybro — CHANGES_REQUESTED** [review]

File: storybook/pages/LanguageCurrencySettingsPage.qml [hunk 6]
```diff
@@ -0,0 +1,113 @@
...
+            contentWidth: parent.width
+
+            languageStore: LanguageStore {
+                readonly property string currentLanguage: "en"
+
+                readonly property ListModel languageModel: ListModel {
+                    ListElement {
+                        locale: "ar"
+                        name: "Arabic"
+                        shortName: "العربية"
```
Comment (noeliaSD) [inline storybook/pages/LanguageCurrencySettingsPage.qml:31]:
```suggestion
                        native: "العربية"
```

File: storybook/pages/LanguageCurrencySettingsPage.qml [hunk 7]
```diff
@@ -0,0 +1,113 @@
...
+                        flag: "🇸🇦"
+                        state: 0
+                    }
+                    ListElement {
+                        locale: "en"
+                        name: "English"
+                        shortName: "English"
```
Comment (noeliaSD) [inline storybook/pages/LanguageCurrencySettingsPage.qml:38]:
```suggestion
                        native: "English"
```

File: ui/app/AppLayouts/Profile/views/AboutView.qml (hunk 5 above)
Comment (iurimatias) [reply ui/app/AppLayouts/Profile/views/AboutView.qml:7]: will disapear after the other PR is merged and this one is rebased

File: ui/StatusQ/src/StatusQ/Controls/StatusBinaryRadioControl.qml (hunk 3 above)
Comment (iurimatias) [reply ui/StatusQ/src/StatusQ/Controls/StatusBinaryRadioControl.qml:1]: We had these functionality duplicated multiple times, it also often had bugs or slightly different designs (due to duplication), this single abstraction makes it consistent and without having to worry about details. (it's also a poc/test for other more complex controls down the line). I don't see GroupBox removing this duplication

Comment (caybro) [reply ui/StatusQ/src/StatusQ/Controls/StatusBinaryRadioControl.qml:1]: Yeah I got the general point; but I don't see anything here that could not be solved using a GroupBox. Also a Control should start with Control, not Column. `property var onToggle` is also something that should be replaced by a signal, etc

**Pull request merged**